tests/expressions_case/tests.py

```diff
@@ -1148,6 +1148,33 @@ def test_m2m_reuse(self):
             lambda x: x[1:]
         )
 
+    def test_group_by_on_empty_case(self):
+        self.assertQuerysetEqual(
+            CaseTestModel.objects.values('string')
+            .annotate(test=Case(output_field=IntegerField()), integer_sum=Sum('integer'))
+            .order_by('string'),
+            [
+                {'string': '1', 'test': None, 'integer_sum': 1},
+                {'string': '2', 'test': None, 'integer_sum': 4},
+                {'string': '3', 'test': None, 'integer_sum': 9},
+                {'string': '4', 'test': None, 'integer_sum': 4},
+            ],
+            transform=lambda x: dict(x)
+        )
+        # Test with default value
+        self.assertQuerysetEqual(
+            CaseTestModel.objects.values('string')
+            .annotate(test=Case(default=Value('empty')), integer_sum=Sum('integer'))
```

Comment:
What if `default` is not a constant but a field reference? e.g. `F('integer')`

Reply:
Good point, forgot about that. We’ll need to handle that case.